airflow/providers/tabular/example_dags/example_tabular.py

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+
+from airflow import DAG
+from airflow.operators.bash import BashOperator
+from airflow.providers.tabular.hooks.tabular import TabularHook
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "email": ["airflow@airflow.com"],
+    "email_on_failure": False,
+    "email_on_retry": False,
+}
+
+dag = DAG("tabular", default_args=default_args, schedule_interval=timedelta(1))
+
+t1 = BashOperator(
+    task_id="templated",
+    bash_command="env",
+    env={"TOKEN": TabularHook().get_token()},
```

Comment:
Isn't this code invoked every time a DAG is parsed?

Reply:
Indeed

Reply:
Hmm, good catch. Let me think about a solution

Reply:
Maybe `user_defined_macros` that will be lazily evaluated?

Reply:
We have macros for connections ` {{ conn.my_conn_id }}`  etc...
but there is also another problem - rendering it means that the token is exposed in the render tab of the UI.

Reply:
Thanks @eladkal for the suggestion. I've changed the code to make use of the connection macro. 

I'm not seeing the actual token in the TI details:
![image](https://user-images.githubusercontent.com/1134248/168841056-528eafe0-b636-4599-b604-d13d5956919a.png)

I do see it in the Rendered template:
![image](https://user-images.githubusercontent.com/1134248/168841168-b6f2c9e8-1f10-4858-bb06-c3c44471ead8.png)

But templating it inside of bash isn't the way to do it. It will be passed using an environment variable, which is nicely obfuscated.

Also, I'm seeing the token because I'm explicitly echoing it:
![image](https://user-images.githubusercontent.com/1134248/168841486-d13ed80c-d744-4d81-ab91-f15bf7ebc98d.png)

It probably will be plugged into the `env_vars` of the SparkSubmit or SparkSQL operator: https://airflow.apache.org/docs/apache-airflow-providers-apache-spark/stable/_api/airflow/providers/apache/spark/operators/spark_submit/index.html

Reply:
> We have macros for connections ` {{  conn.test_conn.login }}` etc... but there is also another problem - rendering it means that the token is exposed in the render tab of the UI.

This is a bug - and it should be redacted there too.